end-to-end-applications/kotlin/food-ordering/app/restate-app/src/main/kotlin/dev/restate/sdk/examples/OrderWorkflow.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
+ *
+ * This file is part of the Restate examples,
+ * which is released under the MIT license.
+ *
+ * You can find a copy of the license in the file LICENSE
+ * in the root directory of this repository or package or at
+ * https://github.com/restatedev/examples/
+ */
+package dev.restate.sdk.examples
+
+import dev.restate.sdk.annotation.Handler
+import dev.restate.sdk.annotation.VirtualObject
+import dev.restate.sdk.examples.clients.PaymentClient
+import dev.restate.sdk.examples.clients.RestaurantClient
+import dev.restate.sdk.kotlin.ObjectContext
+import dev.restate.sdk.kotlin.awakeable
+import dev.restate.sdk.kotlin.runBlock
+import kotlin.time.Duration.Companion.milliseconds
+
+/**
+ * Order processing workflow Gets called for each Kafka event that is published to the order topic.
+ * The event contains the order ID and the raw JSON order. The workflow handles the payment, asks
+ * the restaurant to start the preparation, and triggers the delivery.
+ */
+@VirtualObject
```

Comment:
I think this can be a Service

Reply:
I see it's the same for Java... We can also just leave it this way then. With shared handlers it will anyway become a Virtual Object